src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -186,14 +188,29 @@ private static ProcessModuleCollection GetModules(int processId, bool firstModul
 
                         module.ModuleName = new string(chars, 0, length);
 
-                        length = Interop.Kernel32.GetModuleFileNameEx(processHandle, moduleHandle, chars, chars.Length);
+                        for (; ; )
```

Comment:
`GetModuleFileNameEx` takes a `char[]` and you're passing it `char[]?`, does that work?

Reply:
If the compiler can prove to itself that the `char[]?` is not null.

Reply:
The annotation is required for `chars = null;`. The compiler knows it's not null elsewhere.